### PR TITLE
Libmesh update

### PR DESCRIPTION
**Pull request by roystgnr**
* More smart pointers for internal memory management
* MeshBase and ExodusII support for element sets
* Triangulation improvements:
  * Support for specifying or disallowing boundary refinement
  * Mesh holes can be specified using other Mesh objects
* Mesh stitching is now possible on (serialized) DistributedMesh
* Support for more than 65,535 variables in a System
* Various bug fixes and interface updates
  * Updating deprecated ExodusII API uses
  * Compatibility with PETSc 3.17
  * Fix Eigen use of C++17-deprecated call

* MetaPhysicL updates
  * Don't do potentially different sign comparison in sparsity_trim
  * DynamicSparseNumberArray<long long> comm tests
  * Add missing headers, inline keywords

Nothing super-urgent here; I mostly want the triangulation improvements so I can put in the Moose interfaces to them ASAP, and I want to make sure none of the internal changes are affecting any of the more extensive Moose app tests.

### CODE REVIEW

**Comment by dschwen** [comment]
This also adds metaphysicl std::isinf, right?

File: modules/doc/content/newsletter/2022_05.md
Comment (loganharbour) [inline modules/doc/content/newsletter/2022_05.md:27]: 🥳

**Comment by roystgnr** [comment]
> This also adds metaphysicl std::isinf, right?

/me looks around sheepishly

I just got that into libMesh devel this morning; once the devel->master tests with it pass I'll switch this PR to use that merge commit instead.

**Comment by roystgnr** [comment]
And then there turned out to be more bug fixes, and then there turned out to be a couple more features worth squeezing in, and so much for getting this done on Monday.  Fingers crossed, it should be ready to merge now if it passes CI.

File: framework/src/constraints/EqualValueBoundaryConstraint.C
Comment (dschwen) [inline framework/src/constraints/EqualValueBoundaryConstraint.C:173]: so there is a `allgather_packed_range` specialization that simply does not use this last iterator argument?

Comment (roystgnr) [reply framework/src/constraints/EqualValueBoundaryConstraint.C:173]: Is there?  I admit this wouldn't be the first time I'd written an API and forgotten about it later, but everything I see in communicator.h and parallel_implementation.h requires this argument; the approx_buffer_size argument is the only optional one.

Comment (dschwen) [reply framework/src/constraints/EqualValueBoundaryConstraint.C:173]: No, this was a question. What is the point of the `null_output_iterator<Node>()` here?

Comment (lindsayad) [reply framework/src/constraints/EqualValueBoundaryConstraint.C:173]: To satisfy the API we are calling. https://github.com/libMesh/libmesh/pull/3276 is the libMesh PR that necessitated this change

**Review by lindsayad — APPROVED** [review]

**Comment by lindsayad** [comment]
we may want to merge https://github.com/idaholab/moose/pull/21020 first and hopefully let it get through into devel/master because it's necessary for a PHYSOR workshop model that @jortensi is presenting on Sunday. (without that PR we get a nice seg fault or worse at the end of his simulation)

**Comment by roystgnr** [comment]
Sounds wise to me.  Feel free to hit merge on this whenever that's in, if I'm not paying attention?

(and if CI is still happy with this; if there's more delay then I'll be tempted to slip in another libMesh PR or two...)

File: framework/src/constraints/EqualValueBoundaryConstraint.C
Comment (roystgnr) [reply framework/src/constraints/EqualValueBoundaryConstraint.C:173]: Sorry I misunderstood.  The key here is what we're packing and unpacking.  With most objects the API goal was to have as much flexibility as possible - you could send a set of objects and receive into a vector or vice-versa, for example, and since we had to serialize variable-size data into buffers anyway there'd be no extra overhead.

But with Elem and Node specifically, Alex found this was actually less flexible - we were already relying on unpacking those directly into a mesh, we already didn't support doing anything else, so the iterator requirement was redundant in some cases and it was interfering with our use of more advanced APIs in other cases.  So we got rid of it - the unpack itself does the insertion and we no longer need mesh_inserter_iterator for it.

Comment (dschwen) [reply framework/src/constraints/EqualValueBoundaryConstraint.C:173]: Ok, cool, I see that now in the PR Alex linked to. The specializations are `Packing<Node *>` and `Packing<Elem *>`, not a specialization of `allgather_pached_range`.

**Comment by dschwen** [comment]
#2020 is in master

**Comment by loganharbour** [comment]
Please wait for a review from @milljm, @cticenhour, or myself for these. There's some specific commit ordering required to build these environments for the build boxes and this one won't work with it.

**Comment by lindsayad** [comment]
Can you document this process? Everyone on the CCB should have the requisite knowledge to merge MOOSE PRs without having a specific member of the CCB performing review.

File: modules/tensor_mechanics/src/constraints/NodalFrictionalConstraint.C
Comment (cticenhour) [inline modules/tensor_mechanics/src/constraints/NodalFrictionalConstraint.C:17]: @roystgnr Just updated to `moose-libmesh 2022.05.12 build_0` and this causes a C++17 iterator deprecation warning. How do we want to fix this? We'll likely start getting questions soon about it.

Comment (cticenhour) [reply modules/tensor_mechanics/src/constraints/NodalFrictionalConstraint.C:17]: I'm kinda OK with ignoring the warning for now, as I like another libmesh update immediately even less. But open to whatever you think.

**Comment by roystgnr** [comment]
> There's some specific commit ordering required to build these environments for the build boxes and this one won't work with it.

Chat messages, copied for future reference (at least my own reference - I went to this page first before I remembered that I'd gotten the clarification on Slack), until it's no longer a problem:

> We have a stupid flaw where building the environment requires the change to the conda config to be the last commit
> We have a fix for this, but I just haven’t gotten time to actually taking care of it :smile:

**Pull request merged**
